### PR TITLE
Continue refactoring the auth methods for the hub device and module

### PR DESCRIPTION
**Pull request by brycewang-microsoft**
+ consolidate `ClientAuthenticationWithSharedAccessPolicy` for the hub device and module
+ rename `ModuleAuthenticationWithHsm` to `EdgeModuleAuthenticationWithHsm`

### CODE REVIEW

File: iothub/device/src/Authentication/ClientAuthenticationWithSharedAccessPolicy.cs [hunk 1]
```diff
@@ -49,10 +65,13 @@ public string Key
         /// <summary>
         /// Name of the shared access policy.
         /// </summary>
-        public string PolicyName
+        /// <remarks>
+        /// The name of relevant field in a connection string is "SharedAccessKeyName", so we call the property "KeyName" here.
```
Comment (brycewang-microsoft) [inline iothub/device/src/Authentication/ClientAuthenticationWithSharedAccessPolicy.cs:69]: Although the auth method here is using shared access policy, this property is supposed to be "SharedAccessKeyName" within a connection string. Also, we use "SharedAccessKeyName" everywhere else, so rename it.

File: iothub/device/src/Authentication/HsmAuthentication/EdgeModuleAuthenticationWithHsm.cs [hunk 2]
```diff
@@ -10,13 +10,13 @@ namespace Microsoft.Azure.Devices.Client.HsmAuthentication
     /// <summary>
     /// Authentication method that uses HSM to get a SAS token.
     /// </summary>
-    internal class ModuleAuthenticationWithHsm : ClientAuthenticationWithTokenRefresh
+    internal class EdgeModuleAuthenticationWithHsm : ClientAuthenticationWithTokenRefresh
```
Comment (brycewang-microsoft) [inline iothub/device/src/Authentication/HsmAuthentication/EdgeModuleAuthenticationWithHsm.cs:13]: this is a special case applicable for the edge module only, so rename this class

File: iothub/device/src/Authentication/ClientAuthenticationWithSharedAccessPolicy.cs (hunk 1 above)
Comment (abhipsaMisra) [inline iothub/device/src/Authentication/ClientAuthenticationWithSharedAccessPolicy.cs:69]: This explanation is useful for us internally. However, for the customer this naming justification isn't really required. I think it'll be useful if we point to a connection string format MS doc instead. That'll help users understand which value to pass in here.

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**